src/print.rs

```diff
@@ -0,0 +1,318 @@
+use crate::config::{LanguageSettings, ParserSettings};
+use crate::document::code::{CodeBlock, Line, Source};
+use crate::document::transclusion::Transclusion;
+use crate::document::{CompileError, CompileErrorKind, Document, Node};
+use crate::parser::code::RevCodeBlock;
+use std::collections::HashMap;
+use std::fmt::Write;
+
+/// Formats this `Document` as a string containing the documentation file contents
+pub fn print_docs(document: &Document, settings: &ParserSettings) -> String {
+    let mut output = String::new();
+    for node in &document.nodes {
+        match node {
+            Node::Transclusion(transclusion) => {
+                output.push_str(&print_transclusion(transclusion, false))
+            }
+            Node::Text(text_block) => {
+                output.push_str(&text_block.to_string());
```

Comment:
Why not just
```
writeln!(output, "{}", text_block).unwrap();
```
?

Reply:
You see an advantage here by avoiding the push of a newline a a further line of code?

Reply:
The main point here would to avoid the temporary string allocation implied by `.to_string()`. But generally speaking, standardising on `fmt::Formatter` makes the temporaries go away and makes the code mode composable.

Reply:
Yes, definitely. But see https://github.com/mlange-42/yarner/pull/89#discussion_r571621756

Reply:
Made intermediate print functions use `Write` instead of returning `String`